main/CommandHandler.cpp

```diff
@@ -990,6 +990,106 @@ int setAnalogWrite(const uint8_t command[], uint8_t response[])
 }
 
 
+#include "esp_wpa2.h"
+
+int wpa2EntSetIdentity(const uint8_t command[], uint8_t response[]) {
+  char identity[32 + 1];
+
+  memset(identity, 0x00, sizeof(identity));
+  memcpy(identity, &command[4], command[3]);
+
+  esp_wifi_sta_wpa2_ent_set_identity((const unsigned char*)identity, strlen(identity));
+
+  response[2] = 1; // number of parameters
+  response[3] = 1; // parameter 1 length
+  response[4] = 1;
+
+  return 6;
+}
+
+int wpa2EntSetUsername(const uint8_t command[], uint8_t response[]) {
+  char username[32 + 1];
+
+  memset(username, 0x00, sizeof(username));
+  memcpy(username, &command[4], command[3]);
+
+  esp_wifi_sta_wpa2_ent_set_username((const unsigned char*)username, strlen(username));
+
+  response[2] = 1; // number of parameters
+  response[3] = 1; // parameter 1 length
+  response[4] = 1;
+
+  return 6;
+}
+
+int wpa2EntSetPassword(const uint8_t command[], uint8_t response[]) {
+  char password[32 + 1];
+
+  memset(password, 0x00, sizeof(password));
+  memcpy(password, &command[4], command[3]);
+
+  esp_wifi_sta_wpa2_ent_set_password((const unsigned char*)password, strlen(password));
+
+  response[2] = 1; // number of parameters
+  response[3] = 1; // parameter 1 length
+  response[4] = 1;
+
+  return 6;
+}
+
+int wpa2EntSetCACert(const uint8_t command[], uint8_t response[]) {
+
+  // Add +1 for \0 termination
+  size_t ca_cert_buf_size = (command[3] << 8 | command[4]);
+  uint8_t* ca_cert_buf = (uint8_t*)malloc(ca_cert_buf_size + 1);
+
+  memset(ca_cert_buf, 0x00, ca_cert_buf_size + 1);
+  memcpy(ca_cert_buf, &command[6], ca_cert_buf_size);
+
+  esp_wifi_sta_wpa2_ent_set_ca_cert(ca_cert_buf, ca_cert_buf_size + 1);
+
+  response[2] = 1; // number of parameters
+  response[3] = 1; // parameter 1 length
+  response[4] = 1;
+
+  return 6;
+}
+
+int wpa2EntSetCertKey(const uint8_t command[], uint8_t response[]) {
+
+  // Add +1 for \0 termination
+  size_t client_crt_buf_size = (command[3] << 8 | command[4]);
+  size_t client_key_buf_size = (command[5] << 8 | command[6]);
+  uint8_t* client_crt_buf = (uint8_t*)malloc(client_crt_buf_size + 1);
+  uint8_t* client_key_buf = (uint8_t*)malloc(client_key_buf_size + 1);
```

Comment:
Should this RAM be free'ed somewhere?

Reply:
Our use case is somehow particular since the certificates are usually in flash; after a successful connection is established they *should* not be useful but I'm not 100% sure. 
Since our library resets the module upon attempting the first communication with it I think we can avoid freeing them probably.

Reply:
Ok, I'm still wondering if we should use `realloc` instead of `malloc` here, in case the sketch calls the lib side API twice in a row for whatever reason.